diff --git a/lib/constancy/cli.rb b/lib/constancy/cli.rb
index 8caebe6..43cfee0 100644
--- a/lib/constancy/cli.rb
+++ b/lib/constancy/cli.rb
@@ -67,6 +67,12 @@ General options:
 Options for 'check' command:
   --pull       Perform dry run in pull mode
 
+Options for 'pull' command:
+  --yes        Skip confirmation prompt
+
+Options for 'push' command:
+  --yes        Skip confirmation prompt
+
 USAGE
         exit 1
       end
@@ -124,8 +130,8 @@ USAGE
         when :command
           case self.command
           when 'check'      then Constancy::CLI::CheckCommand.run(self.extra_args)
-          when 'push'       then Constancy::CLI::PushCommand.run
-          when 'pull'       then Constancy::CLI::PullCommand.run
+          when 'push'       then Constancy::CLI::PushCommand.run(self.extra_args)
+          when 'pull'       then Constancy::CLI::PullCommand.run(self.extra_args)
           when 'config'     then Constancy::CLI::ConfigCommand.run
           when 'targets'    then Constancy::CLI::TargetsCommand.run
           when nil          then self.print_usage
diff --git a/lib/constancy/cli/pull_command.rb b/lib/constancy/cli/pull_command.rb
index 07dcc19..617b356 100644
--- a/lib/constancy/cli/pull_command.rb
+++ b/lib/constancy/cli/pull_command.rb
@@ -4,7 +4,7 @@ class Constancy
   class CLI
     class PullCommand
       class << self
-        def run
+        def run(args)
           Constancy::CLI.configure
           STDOUT.sync = true
 
@@ -22,7 +22,7 @@ class Constancy
             puts
             puts "Do you want to pull these changes?"
             print "  Enter '" + "yes".bold + "' to continue: "
-            answer = gets.chomp
+            answer = args.include?('--yes') ? 'yes' : gets.chomp
 
             if answer.downcase != "yes"
               puts
diff --git a/lib/constancy/cli/push_command.rb b/lib/constancy/cli/push_command.rb
index 8c90c3d..47bb906 100644
--- a/lib/constancy/cli/push_command.rb
+++ b/lib/constancy/cli/push_command.rb
@@ -4,7 +4,7 @@ class Constancy
   class CLI
     class PushCommand
       class << self
-        def run
+        def run(args)
           Constancy::CLI.configure
           STDOUT.sync = true
 
@@ -22,7 +22,7 @@ class Constancy
             puts
             puts "Do you want to push these changes?"
             print "  Enter '" + "yes".bold + "' to continue: "
-            answer = gets.chomp
+            answer = args.include?('--yes') ? 'yes' : gets.chomp
 
             if answer.downcase != "yes"
               puts
